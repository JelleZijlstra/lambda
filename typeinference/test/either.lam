type either a b = Left a | Right b in
let x = Left 3 in
let y = Right true in
let f = \e. match e with {
	Left n -> n = 1
	| Right b -> b
} in
(f (Left 3), f)
