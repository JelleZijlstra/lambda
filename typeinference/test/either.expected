%%stdout%%
Result: module
	let __result__ = (false, \e. match e with Left n -> n = 1 | Right b -> b)
end
%%stderr%%
